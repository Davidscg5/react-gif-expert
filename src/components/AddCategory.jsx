import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ( { onNewCategory } ) => {
    
    const [inputValue, setInputValue] = useState('');

    const onInputChanged = ( {target} ) => {

        // console.log(event.target.value);
        setInputValue(target.value);
    }
    
    const onSubmit = ( event ) => {

        // console.log('Hola mundo desde onSubmit');
        event.preventDefault();
        // console.log(setCategories);
        // console.log(inputValue);
        if( inputValue.trim().length <= 1) {
            return;   
        } else {
            onNewCategory( inputValue.trim() );
            setInputValue('');
        }

    }

    return (
        <form onSubmit={ onSubmit } aria-label="form">
            
            <input 
                type="text" 
                placeholder="Buscar GIFs" 
                id="npt-gif" 
                value={ inputValue } 
                onChange={ onInputChanged } 
            />

        </form>
    )

}


AddCategory.propTypes = {

    onNewCategory: PropTypes.func.isRequired,
}