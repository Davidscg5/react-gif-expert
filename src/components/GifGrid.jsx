import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {
  
    //Hook personalizado
    const { images, isLoading } = useFetchGifs( category );

    return (
        <>
            <h3>{ category }</h3>
            {/* <LoadingMessage isLoading={ isLoading }/> */}
            {
                isLoading && ( <h2>Cargando...</h2> )
            }
            <div className='card-grid'>
                {
                    images.map( ( image ) => (
                        <GifGridItem 
                        key={ image.id }
                        // Esparcimos todas las propiedades de 'image'
                        { ...image }
                         />
                    ))
                }
            </div>

        </>
    )
}