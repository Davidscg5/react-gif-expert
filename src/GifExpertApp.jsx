import { React, useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState([ '' ]);

    // console.log(categories);
    const onAddCategory = ( newCategory ) => {

        // console.log('The Office');
        // Convertir el nuevo elemento y los elementos existentes en minúsculas
        const lowerCaseNewCategory = newCategory.toLowerCase();
        const lowerCaseCategories = categories.map((category) => category.toLowerCase());

        // Comprobar si el valor en minúsculas ya está en la lista
        if (lowerCaseCategories.includes(lowerCaseNewCategory)) return;

        // Si no está en la lista, agregarlo en su forma original (sin cambiar mayúsculas/minúsculas)
        setCategories([newCategory, ...categories]);
    }

    return (
        <>
            {/* Título */}
            <h1>GifExpertApp</h1>

            {/* Input, componente independiente */}
            <AddCategory 
                onNewCategory={ (value) => onAddCategory(value) }
                currentCategories={ categories }
            />

            {/* Listado de GIFS */}
            <ol>
                {
                    categories.map( (category) => (
                        <GifGrid 
                            key={ category } 
                            category={ category} />
                    ))
                }
            </ol>

            {/* Gif Item */}
        </>
    )

}