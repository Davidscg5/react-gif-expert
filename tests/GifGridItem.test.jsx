import { GifGridItem } from "../src/components";
import { render, screen } from '@testing-library/react';


describe('App GIF Expert', () => {

    const title = 'Rick y Morty';
    const url = 'https://media2.giphy.com/media/f6aTMoYlqUOEViRiBu/giphy.gif?cid=eee15420xucn80h4bt7ewlplhn9p59fev50p1rwhe372kbmo&ep=v1_gifs_search&rid=giphy.gif&ct=g';

    test('Componente GifGridItem render snapshot', () => { 
        
        const { container } = render( 
            <GifGridItem 
                title={ title } 
                url={ url } 
            /> 
        );
        expect( container ).toMatchSnapshot();
     });


     
    test('Debe mostrar la imagen con la URL y el ALT indicado', () => { 
        
        render( 
            <GifGridItem 
                title={ title } 
                url={ url } 
            /> 
        );
        // screen.debug();
        
        expect( screen.getByRole( 'img').src);
     });
});

