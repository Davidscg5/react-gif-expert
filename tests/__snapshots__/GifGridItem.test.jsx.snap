// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App GIF Expert Componente GifGridItem render snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Rick y Morty"
      src="https://media2.giphy.com/media/f6aTMoYlqUOEViRiBu/giphy.gif?cid=eee15420xucn80h4bt7ewlplhn9p59fev50p1rwhe372kbmo&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Rick y Morty
    </p>
  </div>
</div>
`;
